File: test/fixtures/go_project/editable.go
package main

// ISOLATED TEST FILE - Used exclusively by rename and mv tests.
// Do NOT use symbols from this file in grep, refs, calls, or other read-only tests.

import "fmt"

// EditablePerson is a type for testing rename operations.
type EditablePerson struct {
	Name  string
	Email string
}

// NewEditablePerson creates a new EditablePerson.
func NewEditablePerson(name, email string) *EditablePerson {
	return &EditablePerson{Name: name, Email: email}
}

// Greet returns a greeting message.
func (p *EditablePerson) Greet() string {
	return fmt.Sprintf("Hello, %s", p.Name)
}

// editableCreateSample creates an editable sample for testing replace-function.
func editableCreateSample() *EditablePerson {
	return NewEditablePerson("Original Name", "original@example.com")
}

// editableValidateEmail validates an editable email address.
func editableValidateEmail(email string) bool {
	for _, c := range email {
		if c == '@' {
			return true
		}
	}
	return false
}

// EditableStorage is a storage class for testing method replacement.
type EditableStorage struct {
	data map[string]string
}

// NewEditableStorage creates a new EditableStorage.
func NewEditableStorage() *EditableStorage {
	return &EditableStorage{data: make(map[string]string)}
}

// Save stores a value.
func (s *EditableStorage) Save(key, value string) {
	s.data[key] = value
}

// Load retrieves a value.
func (s *EditableStorage) Load(key string) (string, bool) {
	v, ok := s.data[key]
	return v, ok
}

File: test/fixtures/typescript_project/src/editable.ts
/**
 * This file contains functions used exclusively by replace-function tests.
 * Do not use these symbols in other tests to avoid parallel test interference.
 */

import { User } from './user';

/**
 * Creates an editable sample user for testing.
 */
export function editableCreateUser(): User {
    return new User("Original Name", "original@example.com", 30);
}

/**
 * Validates an editable email address.
 */
export function editableValidateEmail(email: string): boolean {
    return email.includes("@");
}

/**
 * Editable storage class for testing method replacement.
 */
export class EditableStorage {
    private data: Map<string, string> = new Map();

    save(key: string, value: string): void {
        this.data.set(key, value);
    }

    load(key: string): string | undefined {
        return this.data.get(key);
    }
}
